packages/concerto-tools/lib/codegen/fromcto/protobuf/protobufvisitor.js

```diff
@@ -0,0 +1,349 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const debug = require('debug')('concerto-core:jsonschemavisitor');
+const util = require('util');
+
+/**
+ * Convert the contents of a {@link ModelManager} to Proto3 files.
+ *
+ * @private
+ * @class
+ * @memberof module:concerto-tools
+ */
+class ProtobufVisitor {
+
+    /**
+     * Transform a Concerto namespace (with a version) to a package name compliant with Proto 3.
+     * @param {Object} concertoNamespace - the Concerto namespace
+     * @return {Object} a package name compliant with Proto 3
+     * @public
+     */
+    concertoNamespaceToProto3SafePackageName(concertoNamespace) {
+        return `${concertoNamespace.split('@')[0]}.v${concertoNamespace.split('@')[1].replace(/\./ig, '_')}`;
+    }
+
+    /**
+     * Transform a Concerto meta property into a Proto3 field rule.
+     * @param {Object} field - the Concerto meta property
+     * @return {Object} the Proto3 field rule
+     * @public
+     */
+    concertoToProto3FieldRule(field) {
+        if (field.isArray()) {
+            return 'repeated';
```

Comment:
Might be good to add a comment here that arrays are implicitly optional in Proto3 so that it's clear why these two conditions are mutually exclusive.

Reply:
Addressed ✅